modules/hal_gigadevice/Kconfig

```diff
@@ -20,6 +20,34 @@ config HAS_GD32_HAL
 
 if HAS_GD32_HAL
 
+choice GD32_HXTAL_FREQUENCY
+	prompt "High speed external oscillator clock frequency"
+	default GD32_HXTAL_DEFAULT
+	help
+	  Define value of high speed crystal oscillator (HXTAL) in Hz
+	  This value sets the frequency of the oscillator.
+	  Only 8MHz and 25 MHz are supported.
+
+config GD32_HXTAL_DEFAULT
+	bool "Default"
+	help
+	  Use default frequency for HXTAL
+	  This use for SoCs (e.g. gd32f4xx, gd32f3x0, etc ...) and
+	  boards (e.g. GD32VF103 EVAL, GD32E103R_START) that
+	  have HXTAL definitions in firmware.
+
+config GD32_HXTAL_8MHZ
+	bool "8MHz"
+	help
+	  Use 8MHz oscillator for HXTAL
+
+config GD32_HXTAL_25MHZ
+	bool "25MHz"
```

Comment:
Should this setting have some `depends on <SOMETHING>` or `depends on !<SOMETHING>`.

I assume it's mutual exclusive to the `default` choice entry.

Reply:
Enable only for GD32VF103.
No other case is available at this time.